### ISSUE TITLE
Error when filtering with a date

### ISSUE THREAD

**Issue by jibaku** [czartur/django-psql-estimate#2]
When I try to estimate the count of a query set that contains a filter with a date. I get a database error because the query contains a date that is not quoted:

`"articles_article"."post_date" < 2015-02-12 11:19:08.222018`
rather than
`"articles_article"."post_date" < '2015-02-12 11:19:08.222018'`

Any idea on how to handle that case ?

**Comment by jibaku** [comment]
I've tried to quote the date before submitting the query to count_estimate, but the problem is that the added quote is escaped when the query is run. So I end up with `"articles_article"."post_date" < \'2015-02-12 11:19:08.222018\'`

**Comment by artcz** [comment]
That's a good question, thanks for posting the issue; I don't have any ideas on the top of my head, but I'll play with it in some free time, and try to figure it out. I'm keeping this open until we get a reliable solution.

**Comment by prokaktus** [comment]
Hi! It's just not works by design, `query` returns invalid sql.
Source: https://code.djangoproject.com/ticket/17741

Looks like there is no workaround, suggested by Django... 😞 

Maybe it should be noticed in README.

**Comment by jibaku** [comment]
Or maybe by using the QuerySet and it's API to create the query by doing the vars interpolation in Python code. But it's maybe too much work.

**Comment by prokaktus** [comment]
@jibaku do you already did this? I've digged into code and this is not looks like easy task, because Django itself not doing any interpolation. Quoting from the link above:

> Django never actually interpolates the parameters: it sends the query and the parameters separately to the database adapter, which performs the appropriate operations.

But I don't know exactly, how it works.

**Comment by jibaku** [comment]
@prokaktus no, I've never done that. For postgresql it can be done with mogrify I think.
http://initd.org/psycopg/docs/cursor.html?highlight=mogrify#cursor.mogrify

**Comment by prokaktus** [comment]
@jibaku great suggestion! you're right :+1: Just tested it, and all parameters interpolates as needed. If it possible to get query from Django with parameters like in the example from psycopg-docs `"INSERT INTO test (num, data) VALUES (%s, %s)", (42, 'bar')`, it will be great!
